src/omero_upload/library.py

```diff
@@ -0,0 +1,66 @@
+from hashlib import sha1
+import logging
+import os
+from StringIO import StringIO
+import omero.clients
+from omero.gateway import BlitzGateway
+
+log = logging.getLogger(__file__)
+# log.setLevel(logging.DEBUG)
+
+bufsize = 1048576
+
+
+def upload_ln_s(client, filepath, omero_data_dir, mimetype=None):
+    """
+    Simulate an in-place ln-s upload by creating an empty file, replacing it
+    with a symlink, and updating the OriginalFile metadata.
+
+    Requires write access to the omero.data.dir Files directory
+
+    :param client: Connected client object
+    :param filepath: Path to the file to be symlinked into OMERO
+    :param omero_data_dir: The OMERO data directory
+    :param mimetype: The mimetype to be assigned to the file in OMERO
+    :return: The wrapped OriginalFile for the symlinked file
+    """
+    abspath = os.path.abspath(filepath)
+    filedir = os.path.dirname(abspath)
+    filename = os.path.basename(abspath)
+
+    conn = BlitzGateway(client_obj=client)
+
+    log.info('Creating empty OriginalFile placeholder')
+    placeholder = StringIO('')
+    fo = conn.createOriginalFileFromFileObj(
+        placeholder, filedir, filename, 0, mimetype=mimetype)
+    omero_path = os.path.join(
```

Comment:
Probably as a RFE, there could be a check that the folder exists with a warning about the fact this needs to be invoked from the server. There might be some commands from upstream OMERO.cli doing this e.g. in `admin.py`.